class.frame-nonce-preview.php

```diff
@@ -45,15 +52,19 @@ public function is_frame_nonce_valid() {
 			return false;
 		}
 
-		Jetpack::load_xml_rpc_client();
-		$xml = new Jetpack_IXR_Client();
-		$xml->query( 'jetpack.verifyFrameNonce', sanitize_key( $_GET['frame-nonce'] ) );
+		if ( empty( $this->frame_nonce_valid ) ) {
+			Jetpack::load_xml_rpc_client();
+			$xml = new Jetpack_IXR_Client();
+			$xml->query( 'jetpack.verifyFrameNonce', sanitize_key( $_GET['frame-nonce'] ) );
 
-		if ( $xml->isError() ) {
-			return false;
+			if ( $xml->isError() ) {
+				return false;
```

Comment:
Do we want to set `frame_nonce_valid` in this case?

Reply:
We probably do, you're right